cdk/stack.py

```diff
@@ -0,0 +1,100 @@
+from aws_cdk import core
+from aws_cdk import aws_iam as iam
+import os
+from settings import settings
+from permissions_boundary import PermissionsBoundaryAspect
+import os
+import docker
+from aws_cdk import (
+    core,
+    aws_apigateway as apigw,
+    aws_lambda as lambda_,
+)
+from typing import Dict
+
+gcc_permissions_boundary_enabled = os.environ.get('GCC_PERMISSIONS_BOUNDARY_ENABLED', 'False').lower() != 'false'
+
+
+class EptServeTilesStack(core.Stack):
+    def __init__(self, scope: core.Construct, stack_name: str, **kwargs) -> None:
+        super().__init__(scope, stack_name, **kwargs)
+
+        self.gateway = EptServeTilesLambdaGateway(
+            self,
+            id="EptServer",
+            stack_name=stack_name,
+            env=settings.dict(),
+        )
+
+        if gcc_permissions_boundary_enabled:
+            permission_boundary = iam.ManagedPolicy.from_managed_policy_name(
+                self, "PermissionsBoundary", "gcc-tenantOperatorBoundary"
+            )
+            self.node.apply_aspect(PermissionsBoundaryAspect(permission_boundary))
+
+
+class EptServeTilesLambdaGateway(core.Construct):
+
+    def __init__(
+        self,
+        scope: core.Construct,
+        id: str,
+        stack_name: str,
+        memory: int = 512,
+        timeout: int = 30,
+        entry: str = ".",
+        handler: str = "lib/lambda.handler",
+        env: Dict = None,
+        runtime: lambda_.Runtime = lambda_.Runtime.NODEJS_12_X,
```

Comment:
nodejs10.x is stage 1 EOL at the end of July and stage 2 EOL at end of Aug, so upgrading to 12.